### DISCUSSION TITLE
Device Import - Row 1 rack: Object not found

### DISCUSSION THREAD

**Discussion by dennisscg** [netbox-community/netbox#10166] [General]
NetBox version
v3.3.0

Python version
3.8

Expected Behavior
a new device will be created and is assigned to the correct rack

having issues using https://netbox/dcim/devices/import/
although it works with adding the same thing with api.

serial,device_role,manufacturer,device_type,status,site,rack,position,face,cf_CPU_Model,cf_CPUs,cf_Cores,cf_Memory
000000000,server-role,HP,ProLiant DL380 Gen10,active,7,203,27,front,Xeon Silver 4214R 2.4GHz,2,24,0

rack and location exists in netbox.

![netbox](https://user-images.githubusercontent.com/112192127/186947273-2c6340a7-84c1-4018-85b3-e3ebd4e6a0ce.PNG)

**Comment by candlerb** [comment]
I think it's saying it can't find a rack "203", at least, not in the site with name "7" (although it's not complaining about the site - unless you've truncated the screenshot). Are you sure you have a site and rack with those names?

**Reply by dennisscg** [reply]
hey Brian, yes i am sure i have both. It works with API but not via GUI.

**Reply by candlerb** [reply]
And you're sure your site's *name* is 7, not its ID?  Ditto for the rack?

**Reply by dennisscg** [reply]
which variable should i be using for site id ?  

Site
Name
Slug 
id 

? i am currently using Site* 7

**Reply by dennisscg** [reply]
![netbox](https://user-images.githubusercontent.com/112192127/187210164-9cadad47-83e8-4578-9d74-fd0a4214d79d.PNG)

**Reply by candlerb** [reply]
I've always used the site name and rack name.

Looking at [source code](https://github.com/netbox-community/netbox/blob/v3.3.1/netbox/dcim/forms/bulk_import.py#L357-L374) it agrees:

```
class DeviceCSVForm(BaseDeviceCSVForm):
    site = CSVModelChoiceField(
        queryset=Site.objects.all(),
        to_field_name='name',
        help_text='Assigned site'
    )
...
    rack = CSVModelChoiceField(
        queryset=Rack.objects.all(),
        to_field_name='name',
        required=False,
        help_text="Assigned rack (if any)"
    )
```

**Reply by dennisscg** [reply]
so that means what i am using is correct, so maybe its a bug ?

**Reply by candlerb** [reply]
Only other idea I have is about uniqueness: you don't have two racks with name "203", or two sites with name "7" ?

See if you can reproduce it on a clean install, giving the exact set of data to enter (sites and racks).  If you can do this, and provide the steps, then it's a bug.

**Reply by dennisscg** [reply]
just checked again, site and rack are unique, no other rack matched that.

**Reply by ydc-dg** [reply]
Running v3.2.1
And I have the same problem.

**Comment by dennisscg** [comment]
Just figured out the netbox import issue where it kept throwing that error about the rack. Location also needs to be defined. that was my problem.

**Comment by dennisscg** [comment]
serial,device_role,manufacturer,device_type,status,location,site,rack,position,face,cf_CPU_Model,cf_CPUs,cf_Cores,cf_Memory

**Comment by dennisscg** [comment]
to: DEVELOPERS, maybe you can adjust the error msg so it will explain that location is missing instead of rack not found.

Thank you.

**Comment by Glow-Verse** [comment]
Server Error

There was a problem with your request. Please contact an administrator.

The complete exception is provided below:

<class 'dcim.models.racks.Rack.site.RelatedObjectDoesNotExist'>

Rack has no site.

Python version: 3.10.6
NetBox version: 3.5.7

**Reply by candlerb** [reply]
Please don't just post an error message with no context. If you don't describe what you're doing which causes the error, we have no idea what might be the cause.

And unless you can demonstrate with certainty that this is related to the original problem in this thread (i.e. you are importing Devices via CSV), then it would be better start a new thread.
